### PR TITLE
Quest task cleanup

### PR DESCRIPTION
**Pull request by kirderf1**
This pull request does not focus on solving any specific issue or adding features, but rather focuses on internal changes for quest tasks to make them easier to code with, introduce abstract classes for common features, and generally modify the code to better adhere to programming principles.

Notable changes include:
- Result consumers for additional edit menus
- Part list object to replace arrays of task parts
- Organized task classes into packages and renamed classes
- Task drawing and clicking functions separated into new task graphic classes
- Task data uses lists instead of arrays
- Generify tasks to remove need of casting task data

End-user changes include:
- Tame tasks now default icons to mob eggs again
- Soft limits on the number of task parts have been added
There are probably more, but I'd assume that they're less notable

### CODE REVIEW

File: common/src/main/java/hardcorequesting/common/client/interfaces/edit/PickMobMenu.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+import net.minecraft.world.entity.player.Player;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class PickMobMenu extends GuiEditMenuExtended {
+    //TODO add support for entity tags to replace functionality of this special case
```
Comment (canitzp) [inline common/src/main/java/hardcorequesting/common/client/interfaces/edit/PickMobMenu.java:23]: Do you plan to do this or is this TODO for a future contributor?

File: common/src/main/java/hardcorequesting/common/quests/task/TaskType.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import hardcorequesting.common.quests.task.item.*;
+import hardcorequesting.common.quests.task.reputation.HaveReputationTask;
+import hardcorequesting.common.quests.task.reputation.KillReputationTask;
+import hardcorequesting.common.util.EditType;
+import hardcorequesting.common.util.SaveHelper;
+import hardcorequesting.common.util.Translator;
+
+import java.lang.reflect.Constructor;
+
+public enum TaskType {
```
Comment (canitzp) [inline common/src/main/java/hardcorequesting/common/quests/task/TaskType.java:17]: Are you sure to leave this as an enum? I don't know of any mod expanding HQM Tasks, but with this change, future approaches of addon-tasks aren't possible

Comment (kirderf1) [reply common/src/main/java/hardcorequesting/common/quests/task/TaskType.java:17]: For now, yes.
I certainly think that it should be changed to some sort of registry, but I felt that it was out of scope for this specific pull request, so I didn't do more than move this enum out from the already bloated Quest class.

Comment (canitzp) [reply common/src/main/java/hardcorequesting/common/quests/task/TaskType.java:17]: Yeah thats fine, I thought of that

File: common/src/main/java/hardcorequesting/common/client/interfaces/edit/PickMobMenu.java (hunk 1 above)
Comment (kirderf1) [reply common/src/main/java/hardcorequesting/common/client/interfaces/edit/PickMobMenu.java:23]: It's meant as a TODO for a future contributor. I had decided that entity tags was out of scope for this pull request.

**Pull request merged**